rpc/src/module/chain.rs

```diff
@@ -1645,4 +1692,107 @@ impl ChainRpc for ChainRpcImpl {
         );
         Ok(Some(median_time.into()))
     }
+
+    fn get_median_fee_rate(
+        &self,
+        blocks_to_scan: Option<Uint32>,
+        min_transactions: Option<Uint32>,
+    ) -> Result<ckb_jsonrpc_types::Capacity> {
+        let blocks_to_scan = blocks_to_scan
+            .map(|v| v.value())
+            .unwrap_or(MAX_BLOCKS_TO_SCAN);
+        let min_transactions = min_transactions
+            .map(|v| v.value())
+            .unwrap_or(MIN_TRANSACTIONS);
+
+        // check input parameter
+        if blocks_to_scan > MAX_BLOCKS_TO_SCAN {
+            return Err(RPCError::invalid_params(format!(
+                "invalid input parameter: max_blocks_to_scan, should be less than {}",
+                MAX_BLOCKS_TO_SCAN
+            )));
+        }
+        if min_transactions < MIN_TRANSACTIONS {
+            return Err(RPCError::invalid_params(format!(
+                "invalid input parameter: min_transactions, should be greater than {}",
+                MIN_TRANSACTIONS
+            )));
+        }
+
+        // find matched blocks
+        let snapshot = self.shared.snapshot();
+        let tip = snapshot.tip_header().number();
+        let mut match_blocks = Vec::with_capacity(blocks_to_scan as usize);
+        let mut counter = 0;
+
+        // TODO: parallel?
+        for index in (0..tip).rev() {
```

Comment:
This loop may cause a large impact on mainnet nodes. Because there are not many transactions on the present mainnet main-chain.